src/sql/src/rbac.rs

```diff
@@ -70,6 +70,24 @@ macro_rules! rbac_preamble {
     };
 }
 
+/// RBAC requirements for executing a given plan.
+struct RbacRequirements {
```

Comment:
I love the consolidation, a single source of truth for what features are required is great!